### ISSUE TITLE
Pedestrian street with vehicles allowed not routable by car profile

### ISSUE THREAD

**Issue by michaelkrog** [graphhopper/graphhopper#2888]
**Describe the bug**
A small island in Denmark is only accessible by car via a very short pedestrian street that allows vehicles. All routes to most of the island is blocked because graph hoper does not allow vehicle routes through this pedestrian street,

Examples:
[Graphhopper](https://graphhopper.com/maps/?point=55.175706%2C11.648719&point=55.175757%2C11.649549&profile=car&layer=Omniscale) (fails to route)
[Google maps](https://www.google.com/maps/dir/55.1757211,11.6486372/55.1757589,11.6495113/@55.1762306,11.6489416,17.83z/data=!4m2!4m1!3e0?entry=ttu)
[OSRM](https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=55.17643%2C11.64703%3B55.17549%2C11.64949#map=19/55.17600/11.64842)

**To Reproduce**

Steps to reproduce the behavior. For examples:

1. Go to https://graphhopper.com/maps/?point=55.175706%2C11.648719&point=55.175757%2C11.649549&profile=car&layer=Omniscale 
2. Confirm that it cannot create a full route between locations.
3. Go to https://www.openstreetmap.org/way/977919745.
4. Confirm that vehicles are allowed

**Expected behavior**
I expect vehicles are allowed when it is marked as such in OSRM data.

**System Information**

Public graphhopper

**Screenshots & Logs**

<img width="618" alt="Skærmbillede 2023-10-10 kl  10 42 59" src="https://github.com/graphhopper/graphhopper/assets/681472/21e70a8f-d28c-4752-be64-f926ab4de8f2">

**Comment by easbar** [comment]
Shouldn't this rather be tagged as `living_street`?

**Comment by otbutz** [comment]
The translation of `Gågade zone` is literally `pedestrian zone`: https://www.deepl.com/translator#da/en/g%C3%A5gade%20zone

It's even listed in the OSM tagging examples(last one): https://wiki.openstreetmap.org/wiki/Tag:highway=pedestrian?uselang=en

**Comment by otbutz** [comment]
Seems to be pretty common: https://overpass-turbo.eu/s/1BFa

**Comment by easbar** [comment]
Maybe exclude `area=yes`, but true there seem to be quite a few pedestrian highways using `vehicle=yes` (https://overpass-turbo.eu/s/1BFq)

**Comment by otbutz** [comment]
And a bit more if we also take different values like `destination` into account: https://overpass-turbo.eu/s/1BFr

**Comment by otbutz** [comment]
This has also implications for bike routing. If cyclists are explicitly allowed to enter, we shouldn't treat it as a pushing section.

**Comment by easbar** [comment]
It would be nice to see a few more real-world examples where accepting the pedestrian roads is necessary. I know the overpass query returns lots of ways with this tag combination, but looking for some actual examples where this is an issue would make it clear that we really need to change this and we do not break more cases than we fix.

Once this is clear it should be a matter of leaving out `pedestrian` from `import.osm.ignored_highways` (it is currently included in `config-example.yml` and also in the hosted car profile) (this will impact the performance of the OSM import as well as the resulting graph size) and adding `pedestrian` to `highwayValues` in `CarAccessParser.java` (and possibly a few other access parsers).

**Comment by michaelkrog** [comment]
Ok.

I can dig up some more examples from Denmark. It is a pretty common thing here to have pedestrian areas that allows vehicles - also for drive throughs. 

Normally it isn't an issue I guess, because driving around is often more optimal because of the low max speed. 

But for deliveries in the pedestrian zone and drive throughs to reach an "island", it is difficult to do without.

**Comment by easbar** [comment]
> It is a pretty common thing here to have pedestrian areas that allows vehicles - also for drive throughs.

Are they allowed to drive through for anyone or only for delivery vehicles and such?

> and drive throughs to reach an "island", it is difficult to do without.

Yes, of course excluding an entire island because of such a pedestrian road is bad.

**Comment by otbutz** [comment]
> It would be nice to see a few more real-world examples where accepting the pedestrian roads is necessary.

Here's an example from the OSM wiki:

![image](https://github.com/graphhopper/graphhopper/assets/22315436/26549703-7333-40c7-9152-f01d5f77b7b8)

> "Gågade" Pedestrian street with "Kørsel tilladt" vehicles allowed. Driving is allowed here, but (tourist)pedestrians are prioritized higher, so motorists basically can't expect to move faster than the pedestrian in front of them.

**Comment by michaelkrog** [comment]
Adding a few examples here.

## 1: Korsør, Denmark
Ground truth (signs says "driving allowed" and "one way")
<img width="1499" alt="Skærmbillede 2023-10-12 kl  09 57 25" src="https://github.com/graphhopper/graphhopper/assets/681472/65ec83c5-0bc6-43c7-8faa-69a97564976c">

Graphhopper maps route

<img width="926" alt="Skærmbillede 2023-10-12 kl  09 58 02" src="https://github.com/graphhopper/graphhopper/assets/681472/4a12ecd5-ef54-428c-afed-3274fce86d0c">

## 2: Korsør, Denmark
Ground truth (signs says "driving allowed")
<img width="1557" alt="Skærmbillede 2023-10-12 kl  10 06 50" src="https://github.com/graphhopper/graphhopper/assets/681472/23df9ce5-4747-4d72-88e9-68c4dbf09c62">

Destination is parking spots in Pedestrian zone
<img width="1340" alt="Skærmbillede 2023-10-12 kl  10 08 11" src="https://github.com/graphhopper/graphhopper/assets/681472/40dcb755-b128-4bee-9539-ed28839febcc">

Graphhopper maps route
<img width="350" alt="Skærmbillede 2023-10-12 kl  10 07 48" src="https://github.com/graphhopper/graphhopper/assets/681472/8f5243cf-2fa0-41d7-a66d-e361115a02a8">

## 3: Slagelse, Denmark (bicycles allowed)
Ground truth (sign says "cycling allowed")
<img width="1005" alt="Skærmbillede 2023-10-12 kl  10 13 04" src="https://github.com/graphhopper/graphhopper/assets/681472/de7c0272-bce0-4071-bf64-c1dc28f1abd6">

Graphhopper maps (by bicycle)
<img width="786" alt="Skærmbillede 2023-10-12 kl  10 13 10" src="https://github.com/graphhopper/graphhopper/assets/681472/e5033833-61f1-4e3b-9fce-c3f93336465f">

## 4: Slagelse, Denmark (tricky one, this one)
 Ground truth 
 Signs say (in order):
 
 - Errand driving allowed to church. Cycling allowed.
 - goods traffic allowed 0-11
 - taxis allowed
 - Resident driving allowed
 
<img width="1578" alt="Skærmbillede 2023-10-12 kl  10 16 08" src="https://github.com/graphhopper/graphhopper/assets/681472/ab4cf6c7-dbd3-4549-b2ff-a89290379cff">

Graphhopper maps (by van)
<img width="792" alt="Skærmbillede 2023-10-12 kl  10 22 08" src="https://github.com/graphhopper/graphhopper/assets/681472/d188a3b2-6cb9-4b10-8f55-64ad0ac38430">

**Comment by michaelkrog** [comment]
> This has also implications for bike routing. If cyclists are explicitly allowed to enter, we shouldn't treat it as a pushing section.

What do you mean by "pushing section"? To me it seems that it works for bikes when they are allowed in pedestrian zones.

**Comment by otbutz** [comment]
Seems like we handle this case already properly for bike routing:

https://graphhopper.com/maps/?point=55.403963%2C11.35298&point=55.402384%2C11.354302&profile=bike&layer=OpenStreetMap

* `highway=pedestrian` + `bicycle=yes`: https://www.openstreetmap.org/way/333794059 -> driving allowed
* `highway=pedestrian` : https://www.openstreetmap.org/way/10710648#map=19/55.40315/11.35446 -> pushing section

**Comment by michaelkrog** [comment]
Oh. Literally _a section for pushing_ bicycles. 😀 I thought **pushing section** was a technical term in Gaphhopper I didn't know of. I get it know.

**Comment by michaelkrog** [comment]
@easbar 
> this will impact the performance of the OSM import as well as the resulting graph size
Would it make any sense to introduce a tag filter (if none exists already), allowing fx. the car profile to include pedestrian only if it also has specific tags?

**Comment by easbar** [comment]
Yes, we need to allow highway=pedestrian for car only if it is explicitly allowed by other tags. The import performance is probably not too critical since there aren't so many ways tagged as `highway=pedestrian` overall.

**Comment by michaelkrog** [comment]
Okay. What I meant was filtering by tags already during the import process to only process a minimum `highway=pedestrian`. 

I tried to se the difference in nodes/ways for with and without a filter on tags with a bbox of Denmark.

`way["highway"="pedestrian"][~"^(vehicle|motor_vehicle|motorcar)$"~"^(yes|destination)$"]["area"!~"yes"]({{bbox}});`
Nodes: 1424
Ways: 371

`way["highway"="pedestrian"]({{bbox}}) `
Nodes: 20236
Ways: 2898

**Comment by easbar** [comment]
> So as I understand it, only about a tenth is applicable to a car profile

Yes, this way the number of additional edges would be reduced, but really the overall number of pedestrian edges will be small compared to the rest either way (less than 1% of all OSM highways are pedestrian). 

The first place where OSM ways are filtered is in OSMParsers.java#acceptWay. Everything that passes here will produces some edges. If we wanted to apply the optimization you have in mind it would be here, but like I said maybe that's not even necessary. 

We then need to make sure that in CarAccessParser.java#handleWayTags the access of these edges is false unless there is some tag like vehicle=car.

**Comment by michaelkrog** [comment]
@easbar I am gonna take a look at the issue with routing to/through pedestrian streets.

But... Isn't it gonna cause issues in big city centers where routing through pedestrian streets will often be the shortest route?

It is quite common (in Denmark at least) for pedestrian streets to not have a tag for max speed, although common knowledge here is that pedestrians streets have a low advised max speed (15kmh). Some pedestrian streets are =destination which would hinder route through, but not pedestrian streets  has that and it requires correct tagging.

I am afraid that allowing routing through pedestrian streets will cause unwanted routes in many city centers with poor tagging. 🤔

**Comment by Vectorial1024** [comment]
Hi there, I see there are already 2 PRs that try to solve this issue, so should we review this?

I just discovered this topic, and hopefully I have something helpful to add.

Is it possible for GraphHopper to look at whether e.g. `vehicle=yes` exists in the pedestrian street and then allow cars to pass through them? Right now GraphHopper is following the OSM wiki interpretation of pedestrian streets where cars are not expected to go through them, and indeed, blanket-yes accepting `highway=pedestrian` can cause problems.

There are some paths in Hong Kong that are tagged as `highway=pedestrian` but the street itself IRL does allow vehicular traffic (i.e., it has e.g. `vehicle=yes`). Just that in practice, the street is simply usually packed full of pedestrians, and does not reject vehicular traffic. I personally don't quite agree to this tagging, but it do be like this right now.

If GraphHopper can look at `vehicle=yes` and the related tags, then mappers can simply provide `vehicle=yes` on an case-by-case basis to unlock specific pedestrian streets. This also opens up opportunity to make this behavior country-based.

**Comment by karussell** [comment]
This was implemented in #3203 (i.e. only accessible for cars where explicitly allowed)
